### PR TITLE
Add filter to purchase summary

### PR DESCRIPTION
**Pull request by ericnicolaas**
Would be nice to be able to filter the purchase summary (we need to include certain "fees" in the summary sent to the gateway).

### CODE REVIEW

File: includes/cart/functions.php [hunk 1]
```diff
@@ -750,7 +750,7 @@ function edd_get_purchase_summary( $purchase_data, $email = true ) {
 
 	$summary = substr( $summary, 0, -2 );
 
-	return $summary;
+	return apply_filters( 'edd_get_purchase_summary', $summary, $purchase_data );
```
Comment (pippinsplugins) [inline includes/cart/functions.php]: Add in the `$email` parameter here and then we can merge it in.

**Comment by ericnicolaas** [comment]
Done

**Comment by pippinsplugins** [comment]
We'll include it shortly after 2.4

**Comment by ericnicolaas** [comment]
@pippinsplugins I have added one additional modification in here, which is to check for downloads before looping over the array. Without this, you run into a PHP Warning when making a purchase with a cart that only has fees in it (no downloads).

**Comment by pippinsplugins** [comment]
@ericnicolaas Thanks, we'll get this reviewed for 2.4.2

**Pull request merged**